serde_codegen/src/bound.rs

```diff
@@ -79,6 +79,13 @@ pub fn with_bound<F>(
     }
     impl visit::Visitor for FindTyParams {
         fn visit_path(&mut self, path: &ast::Path, _id: ast::NodeId) {
+            if let Some(seg) = path.segments.last() {
+                if seg.identifier.name.as_str() == "PhantomData" {
```

Comment:
While it's not great to implement a custom type called Phantom data, i think this should check the rest of the path, too.